eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/comm/AsyncHttpRequestManager.java

```diff
@@ -0,0 +1,219 @@
+
+  /**
+   * Boolean indicating if debug logging is enabled. This value is used to prevent unnecessary debug
+   * logging and associated string concatenation when disabled.
+   */
+  private static boolean isDebugEnabled = false;
+
+  /**
+   * Initializes the asynchronous HTTP manager if it has not already been initialized. If the
+   * asynchronous HTTP manager has already been initialized, this method will do nothing.
+   *
+   * @param isDebugEnabled boolean indicating if debug logging is enabled.
+   * @since 2.0.0
+   */
+  public static void initialize(boolean isDebugEnabled) {
+    if (isNotInitialized()) {
+      LOGGER.info("Initializing Asynchronous HTTP manager...");
+
+      // Initialize the asynchronous HTTP client
+      IOReactorConfig ioConfig =
+          IOReactorConfig.custom().setIoThreadCount(MAX_CONCURRENT_CONNECTIONS).build();
+      RequestConfig requestConfig =
+          RequestConfig.custom()
+              .setConnectTimeout(CONNECT_TIMEOUT)
+              .setSocketTimeout(SOCKET_TIMEOUT)
+              .build();
+      httpAsyncClient =
+          HttpAsyncClients.custom()
+              .setMaxConnTotal(MAX_CONCURRENT_CONNECTIONS)
+              .setMaxConnPerRoute(MAX_CONCURRENT_CONNECTIONS)
+              .setDefaultIOReactorConfig(ioConfig)
+              .setDefaultRequestConfig(requestConfig)
+              .build();
+      httpAsyncClient.start();
+
+      // Store debug logging enabled status
+      AsyncHttpRequestManager.isDebugEnabled = isDebugEnabled;
+
+      LOGGER.info("Finished Asynchronous HTTP manager initialization.");
+    } else {
+      if (AsyncHttpRequestManager.isDebugEnabled) {
+        LOGGER.debug(
+            "Asynchronous HTTP manager already initialized. Skipping duplicate initialization.");
+      }
+    }
+  }
+
+  /**
+   * Returns a boolean indicating if the asynchronous HTTP manager has not yet been initialized.
+   *
+   * @return true if the asynchronous HTTP manager has not yet been initialized, false otherwise
+   * @since 2.0.0
+   */
+  public static boolean isNotInitialized() {
+    return httpAsyncClient == null;
+  }
+
+  /**
+   * Shuts down the asynchronous HTTP manager. This method should be called when the asynchronous
+   * HTTP manager is no longer needed.
+   *
+   * @since 2.0.0
+   */
+  public static void shutdown() {
+    if (httpAsyncClient != null) {
+      try {
+        httpAsyncClient.close();
+        httpAsyncClient = null;
+      } catch (Exception e) {
+        LOGGER.error("Error shutting down asynchronous HTTP manager.", e);
+      }
+    }
+  }
+
+  /**
+   * Sends the specified HTTP request to the desired Talk2M API asynchronously. The asynchronous
+   * HTTP manager must be initialized before this method is called.
+   *
+   * @param request HTTP request to send
+   * @param callback callback to execute when the request completes
+   * @return {@link Future} object representing the request
+   * @since 2.0.0
```

Comment:
IllegalStateException is an unchecked exception and is therefore not recorded in Javadoc like a FileNotFoundException, etc. would be.

Reply:
What is the rational for not documenting unchecked exceptions? I have seen JAVA docs that record IOExceptions, isn't this another form of an un-checked exception?
![image](https://user-images.githubusercontent.com/72526279/229823995-41470b8f-827e-41f6-a73c-0e5a28882e60.png)

Reply:
IOException is a checked exception in Java. 

Unchecked exceptions typically result from faulty logic or improper usage of the function/method (such as a bad value specified)
- ArithmeticException
- NullPointerException
- IllegalStateException
- IllegalThreadStateException
- etc...
Checked exceptions typically result from errors that are known to possibly occur (such as missing file)
- ClassNotFoundException
- InterruptedException
- IOException
- FileNotFoundException
- etc...

In the past, typically, these are undocumented in Javadocs within our applications, although perhaps we change that.
It might make sense that if we explicitly throw an unchecked exception, we also document it within Javadocs?